04 - Poker.playground/Contents.swift

```diff
@@ -6,18 +6,286 @@
  - Wir modellieren `Card` als Struct, und `Suit` und `Rank` als Enums. Warum verwenden wir keine Klasse für `Card`? Warum eignet sich ein Enum so hervorragend für `Suit` und `Rank`?
 */
 /*
- EURE ANTWORT HIER
+ keine klasse als card <> reference type
 */
 
+import Foundation
+
+enum Suit: Int, CustomStringConvertible {
+    case caro = 1
+    case herz = 2
+    case pik = 3
+    case kreuz = 4
+    
+    var description: String {
+        switch self {
+    
+        case .caro:
+            return "\u{2662}"
+        case .herz:
+            return "\u{2661}"
+        case .pik:
+            return "\u{2660}"
+        case .kreuz:
+            return "\u{2663}"
+      
+        }
+    }
+}
+
+
+
+enum Rank: Int, CustomStringConvertible  {
+    case two = 2, three = 3, four = 4, five = 5, six = 6, seven = 7, eight = 8, nine = 9, ten = 10, jack = 11, queen = 12, king = 13, ace = 14
+    
+    var description: String {
+        switch self {
+            
+        case .two:
+            return "2"
+        case .three:
+            return "3"
+        case .four:
+            return "4"
+        case .five:
+            return "5"
+        case .six:
+            return "6"
+        case .seven:
+            return "7"
+        case .eight:
+            return "8"
+        case .nine:
+            return "9"
+        case .ten:
+            return "10"
+        case .jack:
+            return "J"
+        case .queen:
+            return "Q"
+        case .king:
+            return "K"
+        case .ace:
+            return "A"
+      
+        }
+    }
+}
+
+struct Card: CustomStringConvertible, Equatable {
+    
+    let suit: Suit
+    let rank: Rank
+    
+    var description: String {
+        return suit.description + rank.description
+    }
+    
+    
+}
+
+func ==(lhs: Card, rhs: Card) -> Bool {
+    
+    return lhs.suit == rhs.suit && lhs.rank == rhs.rank
+    
+}
+
+
+
+enum Ranking: Int, CustomStringConvertible {
+    case highcard = 1, pair = 2, twoPair = 3, triplet = 4, fullHouse = 5, quadruple = 6, flush = 7, straight = 8, straightFlush = 9, royalFlush = 10
+    
+    
+    var description: String {
+        switch self {
+            
+        case .highcard:
+            return "High Card"
+        case .pair:
+            return "Pair"
+        case .twoPair:
+            return "2 Pairs"
+        case .triplet:
+            return "Three of a kind"
+        case .fullHouse:
+            return "Full House"
+        case .quadruple:
+            return "Four of a kind"
+        case .flush:
+            return "Krass alter, n Flush!"
+        case .straight:
+            return "Boa, ne Straße"
+        case .straightFlush:
+            return "Straigh Flush! Du Wahnsinns Typ!"
+        case .royalFlush:
+            return "Royal Flush"
+            
+        }
+    }
+}
+
+struct PokerHand {
+    
+    var cards: [Card] = []
```

Comment:
Das kann eine Konstante sein (die Poker Hand verändert sich ja nicht) und braucht auch keinen Anfangswert. Stattdessen in `init` ein Array erstellen und es dann dem Attribut zuweisen.